test/OData.QueryBuilder.Test/ODataQueryCollectionTest.cs

```diff
@@ -315,6 +351,32 @@ public void ODataQueryBuilderList_Filter_With_ReplaceCharacters_KeyValuePairs_Ar
                 .Should().Throw<ArgumentException>().WithMessage("KeyValuePairs is null");
         }
 
+        [Fact(DisplayName = "Filter variable dynamic property int=> Success")]
+        public void ODataQueryBuilderList_Filter_Simple_Variable_DynamicProperty_Success()
+        {
+            string propertyName = "ODataKind.ODataCode.IdCode";
+            
+            var uri = _odataQueryBuilderDefault
+                .For<ODataTypeEntity>(s => s.ODataType)
+                .ByList()
+                .Filter((s,f,_) => ODataProperty.FromPath<int>(propertyName) >= 3)
```

Comment:
Confuses that the type is int, and propertyName can be anything

Reply:
The type has to match the type defined in the OData model. Since it's dynamically resolved we cannot now the type so it has to be specified by the programmer. 
Property name can be anything - that's the nature of dynamically resolving properties. 
Anyway I've added validation of the model type.

Reply:
BTW: EF Core did implement similar feature with [`EF.Property<T>()`](https://docs.microsoft.com/de-de/dotnet/api/microsoft.entityframeworkcore.ef.property?view=efcore-6.0)